cmd/prometheus/uname_linux_int8.go

```diff
@@ -17,9 +17,14 @@
 package main
 
 func charsToString(ca []int8) string {
-	s := make([]byte, len(ca))
-	for i, c := range ca {
-		s[i] = byte(c)
+	s := make([]byte, 0, len(ca))
+	for _, c := range ca {
+
+		if byte(c) != 0 {
+			s = append(s, byte(c))
+		} else {
+			break
+		}
 	}
-	return string(s[0:len(ca)])
+	return string(s[:])
```

Comment:
Just `return string(s)` is enough here. `[:]` takes a slice of the given array or slice, but as `s` is the slice we want already, that's not necessary.